adapters/repos/db/lsmkv/segment_serialization_benchmarks_test.go

```diff
@@ -63,3 +71,310 @@ func BenchmarkCollectionNodeKeyIndexAndWriteTo(b *testing.B) {
+func runPreloadBufferAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	fileContents, err := os.ReadFile(tempFile) // File read before benchmark timing.
+	if err != nil {
+		b.Fatal("Failed to read file:", err)
+	}
+
+	reader := bytes.NewReader(fileContents)
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		reader.Reset(fileContents) // Resets reader instead of allocating a new one.
+		err = ParseReplaceNodeIntoPread(reader, uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+// runBufferedFileAccess measures performance of a buffered approach where the entire file is read into memory first,
+// then parsed from a memory buffer. Includes the cost of reading the file into memory.
+func runBufferedFileAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	file, cleanup := openFile(b, tempFile)
+	defer cleanup()
+
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if _, err := file.Seek(0, 0); err != nil {
+			b.Fatalf("Failed to seek file: %v", err)
+		}
+
+		all, err := io.ReadAll(file) // Reading full file into memory.
+		if err != nil {
+			b.Fatal("error reading file:", err)
+		}
+
+		if len(all) == 0 {
+			b.Fatalf("file is empty")
+		}
+
+		err = ParseReplaceNodeIntoPread(bytes.NewReader(all), uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+// runDirectFileAccess measures performance of direct file access using multiple `pread` calls.
+// Includes the overhead of system calls for reading from disk.
+func runDirectFileAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	file, cleanup := openFile(b, tempFile)
+	defer cleanup()
+
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if _, err := file.Seek(0, 0); err != nil {
+			b.Fatal("Failed to seek file:", err)
+		}
+
+		err := ParseReplaceNodeIntoPread(file, uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+func generateTestData(valueSize, keySize, secondaryKeysCount, secondaryKeySize int) ([]byte, error) {
+	buffer := new(bytes.Buffer)
+	var err error
+
+	if err = binary.Write(buffer, binary.LittleEndian, byte(rand.Intn(2))); err != nil {
+		return nil, fmt.Errorf("error writing tombstone binary: %w", err)
+	}
```

Comment:
```suggestion
func generateTestData(b *testing.B, valueSize, keySize, secondaryKeysCount, secondaryKeySize int) ([]byte, error) {
        b.Helper()
        
	var buffer bytes.Buffer
	var err error

        err = binary.Write(buffer, binary.LittleEndian, byte(rand.Intn(2))
        require.NoError(b, err, "writing tombstone binary")
```

suggestion: calling [`b.Helper()`](https://pkg.go.dev/testing#B.Helper) marks this function as a test helper and produces better stack traces in test errors.
This also means that you can "fail" your tests directly in the setup code and still get a clear idea of where exactly the error happened (e.g. during test case No. 3) 

suggestion-2: one of the cool things about Go zero values is that things like `bytes.Buffer` can be used without explicit initialization (quoting Rob Pike again: ["Make the zero value useful"](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=385s))

Reply:
Downgraded that `suggestion-2` to `nit` 🔍